bot/exts/evergreen/duck_game.py

```diff
@@ -0,0 +1,329 @@
+
+        return self._solutions
+
+
+class DuckGamesDirector(commands.Cog):
+    """A cog for running Duck Duck Duck Goose games."""
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+        self.current_games = {}
+
+    @commands.group(name='duckduckduckgoose', aliases=['dddg', 'duckgoose'], invoke_without_command=True)
+    @commands.cooldown(rate=1, per=2, type=commands.BucketType.channel)
+    async def start_game(self, ctx: commands.Context) -> None:
+        """Generate a board, send the game embed, and end the game after a time limit."""
+        if ctx.channel.id in self.current_games:
+            return
+
+        minimum_solutions, = random.choices(range(len(SOLN_DISTR)), weights=SOLN_DISTR)
+        game = DuckGame(minimum_solutions=minimum_solutions)
+        game.running = True
+        self.current_games[ctx.channel.id] = game
+
+        game.embed_msg = await self.send_board_embed(ctx, game)
+        await asyncio.sleep(GAME_DURATION)
+
+        """Checking for the channel ID in the currently running games is not sufficient.
+           The game could have been ended by a player, and a new game already started in the same channel.
+        """
+        if game.running:
+            try:
+                del self.current_games[ctx.channel.id]
+                await self.end_game(game, end_message="Time's up!")
+            except KeyError:
+                pass
+
+    @commands.Cog.listener()
+    async def on_message(self, msg: discord.Message) -> None:
+        """Listen for messages and process them as answers if appropriate."""
+        if msg.author.bot:
+            return
+
+        channel = msg.channel
+        if channel.id not in self.current_games:
+            return
+
+        game = self.current_games[channel.id]
+        if msg.content.strip().lower() == 'goose':
+            # If all of the solutions have been claimed, i.e. the "goose" call is correct.
+            if len(game.solutions) == len(game.claimed_answers):
+                try:
+                    del self.current_games[channel.id]
+                    game.scores[msg.author] += CORRECT_GOOSE
+                    await self.end_game(game, end_message=f"{msg.author.display_name} GOOSED!")
+                except KeyError:
+                    pass
+            else:
+                await msg.add_reaction(EMOJI_WRONG)
+                game.scores[msg.author] += INCORRECT_GOOSE
+            return
+
+        # Valid answers contain 3 numbers.
+        if not (match := re.match(ANSWER_REGEX, msg.content)):
+            return
+        answer = tuple(sorted(int(m) for m in match.groups()))
+
+        # Be forgiving for answers that use indices not on the board.
+        if not all(0 <= n < len(game.board) for n in answer):
+            return
+
+        # Also be forgiving for answers that have already been claimed (and avoid penalizing for racing conditions).
+        if answer in game.claimed_answers:
```

Comment:
I'm wondering if it would be worth adding a reaction to this to indicate that it was reused, probably something different to the cross as they didn't lose a point. Maybe `:recycle:`?

Reply:
My reasoning for this is the same as here: https://github.com/python-discord/sir-lancebot/pull/786#discussion_r692440022